app/src/main/java/io/github/dkter/aaaaa/AaaaaKeyboardView.kt

```diff
@@ -96,17 +104,16 @@ class AaaaaKeyboardView(
         if (this.getBooleanPref(R.string.hapticFeedbackKey))
             v.performHapticFeedback(HapticFeedbackConstants.VIRTUAL_KEY)
 
-        if (id == R.id.btnA) {
-            this.keyboardListener.onA()
-        }
-        else if (id == R.id.btnBackspace) {
-            this.keyboardListener.onBackspace()
-        }
-        else if (id == R.id.btnSpace) {
-            this.keyboardListener.onSpace()
-        }
-        else if (id == R.id.btnReturn) {
-            this.keyboardListener.onReturn()
+        when (id) {
+            R.id.btnA -> keyboardListener.onA()
+            R.id.btnBackspace -> keyboardListener.onBackspace()
+            R.id.btnSpace -> keyboardListener.onSpace()
+            R.id.btnReturn -> keyboardListener.onReturn()
+            R.id.btnUppercase -> {
+                isUppercase = !isUppercase
+                btnA.text = if (isUppercase) "A" else "a"
```

Comment:
I was able to get the caps lock button tint to work by using [`MaterialColors.getColor`](https://developer.android.com/reference/com/google/android/material/color/MaterialColors#getColor(android.content.Context,%20int,%20int)) (you'll need to add `implementation 'com.google.android.material:material:1.3.0'` to build.gradle dependencies).

Reply:
I tried this (c46bc7f) but it throws an exception becuase `R.attr.colorControlNormal` isn't defined in the app's theme.

Reply:
oh, this is weird. I tried it again and `getColor` works when I give it a `defaultValue` (it doesn't use the defaultValue) but when I call it with `errorMessageComponent` it errors. also I noticed that this always uses the light theme version of `colorControlNormal` (even with the dark theme set) so this probably isn't the best solution.

...

After playing around with it I finally found a solution: make the ContextThemeWrapper on line 67 an instance-level attribute and then use this
```
val colorArray = themeWrapper.getTheme().obtainStyledAttributes(intArrayOf(R.attr.colorControlNormal))
val colorControlNormal = colorArray.getColor(0, 0)
colorArray.recycle()
colorControlNormal
```

Reply:
I tried to implement your solution like this (8c00933):
```kotlin
val themeWrapper = ContextThemeWrapper(context, AppCompatDelegate.getDefaultNightMode())
val colorControlNormal = themeWrapper
                .theme
                .obtainStyledAttributes(intArrayOf(android.R.attr.colorControlNormal))
                .use { it.getColor(0, 0) }
```

It works exactly like yours except I don't know how exactly you would've initialized the themeWrapper.

But it seems to always use the light theme version as well. :/

Reply:
For the theme to be correct you need to use [this](https://github.com/dkter/aaaaa/blob/master/app/src/main/java/io/github/dkter/aaaaa/AaaaaKeyboardView.kt#L63) theme wrapper, not create a new one (because this one is manually being set to the correct theme in `init`).

Reply:
Also, [docs](https://developer.android.com/reference/kotlin/android/content/res/TypedArray) say to recycle the `TypedArray` after using it.

Reply:
Don't worry, `use` implicitly calls `recycle()`. 

I'll use that wrapper - I didn't see it - thanks for the help!

Reply:
Resolved in 1d8ebd0.